examples_tests/31.SkinningDataBenchmark/main.cpp

```diff
@@ -4,396 +4,505 @@
+    shaders[1][0] = IAsset::castDown<ICPUSpecializedShader>(vertexShaderBundle_2.getContents().begin()->get());
+    shaders[1][1] = IAsset::castDown<ICPUSpecializedShader>(fragShaderBundle.getContents().begin()->get());
+    shaders[2][0] = IAsset::castDown<ICPUSpecializedShader>(vertexShaderBundle_3.getContents().begin()->get());
+    shaders[2][1] = IAsset::castDown<ICPUSpecializedShader>(fragShaderBundle.getContents().begin()->get());
+    shaders[3][0] = IAsset::castDown<ICPUSpecializedShader>(vertexShaderBundle_4.getContents().begin()->get());
+    shaders[3][1] = IAsset::castDown<ICPUSpecializedShader>(fragShaderBundle.getContents().begin()->get());
+
+    auto createMeshBufferFromGeometryCreatorReturnData = [&](asset::IGeometryCreator::return_type& geometryObject, ICPUMeshBuffer* meshBuffer, uint16_t boneID)
+    {
+        asset::SBlendParams blendParams;
+        asset::SRasterizationParams rasterParams;
+        rasterParams.faceCullingMode = asset::EFCM_NONE;
 
-    const size_t perInstance = sizeof(float)*_boneCnt*4*6;
+        for (int i = 0; i < SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT; i++)
+            meshBuffer->setVertexBufferBinding(std::move(geometryObject.bindings[i]), i);
+        meshBuffer->setBoundingBox(geometryObject.bbox);
+        meshBuffer->setIndexCount(geometryObject.indexCount);
+        meshBuffer->setIndexType(geometryObject.indexType);
+        meshBuffer->setIndexBufferBinding(std::move(geometryObject.indexBuffer));
+        
+        auto newInputParams = geometryObject.inputParams;
+        newInputParams.enabledAttribFlags |= 0b10000u;
+        newInputParams.enabledBindingFlags |= 0b10u;
+        newInputParams.bindings[1].inputRate = EVIR_PER_VERTEX;
+        newInputParams.bindings[1].stride = 0u;
+        newInputParams.attributes[4].binding = 1u;
+        newInputParams.attributes[4].format = EF_R16_UINT;
+        newInputParams.attributes[4].relativeOffset = 0u;
 
-    for (size_t i = 1u; i < _instCnt; ++i)
-        memcpy(_dst + i*perInstance, _dst, sizeof(float)*perInstance);
+        SBufferBinding<ICPUBuffer> boneIDBuffer;
+        boneIDBuffer.offset = 0u;
+        boneIDBuffer.buffer = core::make_smart_refctd_ptr<ICPUBuffer>(meshBuffer->calcVertexCount() * sizeof(uint16_t));
 
-    return sizeof(float) * perInstance * _instCnt;
-}
-size_t convertBuf2(const float* _src, float* _dst, const size_t _boneCnt, const size_t _instCnt)
-{
-    // by mat3x4 (3 rows, 4 columns, column-major layout) i mean glsl's mat4x3...
-    float mat3x4[16]; // layout: m00|m10|m20|---|m01|m11|m21|---|m02|m12|m22|---|m03|m13|m23|--- ("---" is unused padding)
-    float mat3[12];
-    const size_t size_mat3x4_src = 12u;
+        uint16_t* buffPtr = static_cast<uint16_t*>(boneIDBuffer.buffer->getPointer());
+        for (int i = 0; i < meshBuffer->calcVertexCount(); i++)
+            *(buffPtr + i) = boneID;
```

Comment:
need more than one bone per mesh, need artificial duplication of bones, and need a few distinct bones (1-9) being used by different groups of vertices.